server/src/test/java/org/apache/uniffle/server/ShuffleServerMetricsTest.java

```diff
@@ -150,7 +151,8 @@ public void testGrpcMetrics() throws Exception {
 
   @Test
   public void testServerMetricsConcurrently() throws Exception {
-    ExecutorService executorService = Executors.newFixedThreadPool(3);
+    ExecutorService executorService = Executors.newCachedThreadPool(
```

Comment:
This is a temporary thread pool. I think it's unnecessary to be a daemon thread pool.WDYT?